### DISCUSSION TITLE
MyST support

### DISCUSSION THREAD

**Discussion by meghprkh** [remarkjs/remark#1351] [Q&A]
Hi, does anyone know if there are plugins or a set of plugins which enable [myst](https://mystmd.org) support?

Some of the syntax is quite commonmark+extensions friendly, but slightly different. For example, the syntax for admonitions is:

```
:::{note}
abcd
```

But this does not work with remark-directives as it expects the directive as according to it, this is an attribute not an element name.

**Comment by ChristianMurphy** [comment]
Welcome @meghprkh! 👋 
There are not any yet, the only implementation is in Python https://github.com/executablebooks/MyST-Parser
You are welcome to port that to TypeScript/JavaScript if you want
https://github.com/micromark/micromark?tab=readme-ov-file#creating-a-micromark-extension

**Comment by meghprkh** [comment]
Thanks @ChristianMurphy for your reply. To add there is also a JS package based on markdown-it https://github.com/jupyter-book/mystmd/tree/main/packages/markdown-it-myst . I read the comparison in the readme and the pros/cons make sense to me.

I found the remark CLI does linting and formatting which is why I was particularly interested in it.

Should this be a micromark extension or a remark extension? I read remark extension is easier to make on the [README](https://github.com/micromark/micromark?tab=readme-ov-file#micromark). I assume if I write a micromark extension it works with the remark CLI tools?

Currently I do not actively have the time to contribute here, and therefore but if you feel its worth to re-open the discussion, please do so. Hopefully sometime in the future, someone (or me) will come up with a plugin for it :)

**Reply by ChristianMurphy** [reply]
> Should this be a micromark extension or a remark extension? 

Both, micromark is the parser.
It takes text and turns it into tokens.
the remark part takes the tokens and turns them into an AST for easier interactions with users and other plugins.

> I read remark extension is easier to make on the [README](https://github.com/micromark/micromark?tab=readme-ov-file#micromark). I assume if I write a micromark extension it works with the remark CLI tools?

A remark only plugin could work, but, if there's any syntax overlap where you need to override/do something different for Myst. The micromark level is where that needs to happen.
